src/components/specimen/Specimen.tsx

```diff
@@ -85,25 +82,21 @@ const Specimen = () => {
                 }
             });
         }
-    }, [params, version]);
+    }, [specimen, params, version, dispatch]);
 
     /* Onchange of the Annotation Target's annotations: Check if changes occured */
     useEffect(() => {
         /* Check if the specimen annotations differ from the target annotations */
-        if (Array.isArray(annotateTarget.annotations) &&
-            JSON.stringify(specimenAnnotations[annotateTarget.property]) !== JSON.stringify(annotateTarget.annotations)) {
-            CheckAnnotations(specimen);
-        }
-    }, [annotateTarget.annotations]);
-
-    /* Function for checking Specimen Annotations */
-    const CheckAnnotations = (specimen: SpecimenType) => {
-        GetSpecimenAnnotations(specimen.id).then((annotations) => {
-            if (annotations) {
-                dispatch(setSpecimenAnnotations(annotations));
-            }
-        });
-    }
+        // if (Array.isArray(annotateTarget.annotations) &&
```

Comment:
Is this commented piece needed?

Reply:
I recall I commented this, but not sure, anyways uncommented this code, will look if things are working fine when everything comes together